### DISCUSSION TITLE
Media Management Spike RFC

### DISCUSSION THREAD

**Discussion by jamespohalloran** [tinacms/tinacms#3418] [Ideas]
We're starting to plan a spike on making some upgrades to media management, and are looking to surface the highest value media tickets.

Here are some requested upgrades:

- [x] Editors can't create folders in media manager
- [x] UI Grid-view of media items
- Image Drag & Drop always uploads image to the root directory
- Include media metadata in UI & API:
	  - https://github.com/tinacms/tinacms/discussions/3281
	  - https://github.com/tinacms/tinacms/discussions/3419
- Sorting on metadata
- Media Search
- Support for non-image types:
	 - [x] https://github.com/tinacms/tinacms/discussions/3416
	  - https://github.com/tinacms/tinacms/discussions/3417?converting=1 
- Ability to rename files in media manager
- [x] Access direct link to a file in UI. https://github.com/tinacms/tinacms/discussions/2665
- Improved media pagination
          - https://github.com/tinacms/tinacms/discussions/3189
- Capping media file sizes for repo-based media 
          
We likely won't get to all of these in one project, so if anything in this list is extra important to you, or I'm missing anything, please let us know!

**Comment by mattferderer** [comment]
3417, 2665 & 3416 are most important to me. If I can help, I'm glad to try and put some effort in.

**Comment by rabelais88** [comment]
- for me it's media search and metadata. I am using it for [my blog](https://sungryeol.com/) already and it has many image resources. without proper search and metadata, managing images feels like finding a needle in a haystack.
- metadata also needs to include uploaded date.
- also I've noticed external images with markdown is unable to use at current stage. the markdown renderer changes all external image links to internal images. what if there's a licensed image that I can't host at my site?

**Comment by gsicvj** [comment]
Hey people, what is the progress or timeline for the search component in TinaCMS media manager?
I just did a quick research and this is the closest page I could find on the topic.

**Comment by appinteractive** [comment]
https://github.com/tinacms/tinacms/discussions/3419 is Important for me, also to be able to add custom metadata to files like a [Blurhash](https://blurha.sh/) on upload, so I don't need to hack it at build time.

**Comment by Revenciel** [comment]
#3417 is especially important to me--I'd like to be able to upload videos. I'd also like to be able to set default alt-text for images with the media manager, instead of the only option being to set it again every time I insert the image onto a page.

**Reply by matt-antone** [reply]
yes. this would be a huge update

**Comment by mlippe** [comment]
Just checkin in, what the status on this? I would love to upload videos via Tina. Maybe related to #3417
